components/list_modal.tsx

```diff
@@ -1,87 +1,46 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
 import LoadingScreen from 'components/loading_screen';
+import {Group} from '@mattermost/types/groups';
 
 export const DEFAULT_NUM_PER_PAGE = 50;
 
-export default class ListModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * loadItems is a function that receives the params (pageNumber, searchTerm) and should return an object
-         * with the shape {items: [], totalCount: 0}.
-         * items: an array of objects that are passed to each renderRow function.
-         * totalCount: an integer representing the total number of items as displayed in the pagination text.
-         *
-         * Example:
-         *     const loadItems = async (pageNumber, searchTerm) => {
-         *         const {data} = await loadFromServer(searchTerm, pageNumber, PER_PAGE);
-         *         return {
-         *             items: data.users,
-         *             totalCount: data.total,
-         *         };
-         *     };
-         */
-        loadItems: PropTypes.func.isRequired,
-
-        /**
-         * renderRow is a function that receives the params (item, listModal) and should return JSX.
-         * item: an object as returned by each entry in the loadItems function's 'items' array.
-         * listModal: the instance of the ListModal component class.
-         *
-         * Example:
-         *     const renderRow = (item, listModal) => <div>{item.id}</div>;
-         */
-        renderRow: PropTypes.func.isRequired,
-
-        /**
-         * onHide (optional) a function to be invoked when the modal is closed.
-         */
-        onHide: PropTypes.func,
-
-        /**
-         * titleText (optional) a string to show at the top bar of the modal.
-         */
-        titleText: PropTypes.string,
-
-        /**
-         * searchPlaceholderText (optional) a string to show as a placeholder in the search input.
-         */
-        searchPlaceholderText: PropTypes.string,
-
-        /**
-         * titleBarButtonText (optional) a string representing a title bar button text.
-         */
-        titleBarButtonText: PropTypes.string,
-
-        /**
-         * titleBarButtonOnClick (optional) a func to handle title button bar clicks.
-         */
-        titleBarButtonOnClick: PropTypes.func,
-
-        /**
-         * numPerPage (optional) a number setting how many items per page should be displayed. Defaults to
-         * DEFAULT_NUM_PER_PAGE.
-         */
-        numPerPage: PropTypes.number,
-
-        /**
-         * show (optional) a boolean setting to hide the modal via props rather then unmounting it.
-         */
-        show: PropTypes.bool,
-    };
+type Props = {
+    titleText: string;
```

Comment:
Based on the comments, also `titleText`, `show` and `searchPlaceholderText` are optional. Below they are marked as required.